### PR TITLE
Feature/move to netmiko

### PR DESCRIPTION
**Pull request by SimplicityGuy**

### CODE REVIEW

File: pynoc/cisco.py [hunk 1]
```diff
@@ -79,68 +84,53 @@ def connect(self):
...
-            self._send_command(self.CMD_ENABLE, self.CMD_ENABLE_SIGNALS)
-            self._send_command(
-                password, self.CMD_GENERIC_SIGNALS, log=False
-            )
-            self._ready = True
-            self.set_terminal_length()
+        if self.connected:
+            self.disconnect()
+
+        self._client = ConnectHandler(
```
Comment (davidhousedev) [inline pynoc/cisco.py]: Why does a `ConnectionHandler` get instantiated twice? Once, earlier at line 87.

File: pynoc/cisco.py [hunk 2]
```diff
@@ -330,45 +296,35 @@ def _shorthand_port_notation(self, port):
...
-        send_command = command + '\n'
-        if log:
-            self._logger.info('Sending command: %s', send_command.strip())
+        read_buffer = ""
+        if self._client is None:
+            return read_buffer
 
-        self._shell.send(send_command)
+        self._client.clear_buffer()
+        read_buffer = self._client.send_command(command)
```
Comment (davidhousedev) [inline pynoc/cisco.py:308]: Nice. This is so much cleaner. I think this means we can remove the dependency on `six`.

File: requirements.txt [hunk 3]
```diff
@@ -1,5 +1,5 @@
 pysnmp
 netaddr
-paramiko
+netmiko
 retrying
 six
```
Comment (davidhousedev) [inline requirements.txt]: I think we can get rid of this dependency now. I don't remember calling it anywhere other than `CiscoSwitch`.

File: pynoc/cisco.py [hunk 4]
```diff
@@ -1,10 +1,32 @@
...
+import warnings
 
-from six import PY3
 from netaddr import EUI, mac_unix_expanded
-from paramiko import SSHClient, AutoAddPolicy
+from netmiko import ConnectHandler
+
+
+def deprecated(func):
+    """Deprecated decorator.
```
Comment (davidhousedev) [inline pynoc/cisco.py]: Phrasing of this docstring is failing the build at FLAKE8: `pynoc/cisco.py:10:1: D401 First line should be in imperative mood; try rephrasing`. Perhaps rephrase to "Decorates a deprecated function to warn when it is called."

Comment (SimplicityGuy) [reply pynoc/cisco.py]: Didn't hit this originally on my machine due to outdated `flake8-docstrings`. Fixing now.

File: pynoc/cisco.py (hunk 1 above)
Comment (SimplicityGuy) [reply pynoc/cisco.py]: The difference here is the `secret`.  However I'll change this to use the property instead.

File: pynoc/cisco.py [hunk 5]
```diff
@@ -104,14 +104,10 @@ def enable(self, password):
...
-            self.disconnect()
-
-        self._client = ConnectHandler(
-            device_type='cisco_ios', ip=self._host, username=self._username,
-            password=self._password, secret=password
-        )
+        if not self.connected:
+            return
 
+        self._client.secret = password
```
Comment (davidhousedev) [inline pynoc/cisco.py]: Nice

**Review by davidhousedev — APPROVED** [review]

**Pull request merged**